phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/formats/hpo/category/HpoCategoryMap.java

```diff
@@ -21,243 +21,275 @@
+      for (TermId p : parents) {
+        if (this.categorymap.containsKey(p)) {
+          builder.add(p);
+        } else {
+          stack.add(p);
         }
-        return builder.build();
+      }
     }
+    return builder.build();
+  }
 
 
-    /** Identify the category that best matches the term id (usually just fine the category that represents a parent term). */
-    private HpoCategory getCategory(TermId tid, Ontology ontology) {
-        //logger.trace(String.format("GetCategory for %s[%s]",ontology.getTermMap().get(tid).getName(),tid.getIdWithPrefix()));
-        List<HpoCategory> activeCategoryList = new ArrayList<>();
-        if (tid==null) {
-          System.err.println("Warning, tid was null...");
-            return null;
-        }
-        Set<TermId> ancs = getAncestorCategories(ontology,tid);
-        return getPrioritizedCategory(ancs);
+  /**
+   * Identify the category that best matches the term id (usually just
+   * find the category that represents an ancestor term).
+   */
+  private HpoCategory getCategory(TermId tid, Ontology ontology) {
+    if (tid == null) {
+      System.err.println("Warning, tid was null...");
+      return null;
     }
+    Set<TermId> ancs = getAncestorCategories(ontology, tid);
+    return getPrioritizedCategory(ancs);
+  }
 
-
-    private HpoCategory getPrioritizedCategory(Set<TermId> catlist) {
-        for (TermId hpocat : catlist) {
-            if (hpocat.equals(NEOPLASM_ID))
-                return categorymap.get(hpocat);
-        }
-        // if we get here there was no prioritized category
-        // return the first item.
-      return categorymap.values().iterator().next();
+  /**
+   * This method is intended to be used by views that show HPO terms according to top level
+   * categories such as "Skeletal", "Cardiovascular", etc. Because of the multiple parentage of the
+   * HPO, some HPO may have more than one upper level category. For instance, Uveal melanoma would
+   * have both Neoplasm, and Ophthalmology. We prioritize Neoplasm first. For other terms, if there
+   * are multiple parents, we choose one at random for the display. The great majority of terms
+   * will have just one parent, as so we check this first.
+   * The input argument catlist represents the top level categories that some term is  annotated
+   * to. It should never be empty, and will usually just contain one term. The entry/entries in catlist
+   * should all be members of {@link #termIdList}.
+   * We assume that catlist is never empty, which is justified for this private methods because of the
+   * calling code.
+   *
+   * @param catlist A list of HPO TermIds that correspond to the top level categories (see
+   * @return
+   */
+  private HpoCategory getPrioritizedCategory(Set<TermId> catlist) {
+    if (catlist.size() == 1) {
+      return categorymap.get(catlist.iterator().next());
+    }
+    if (catlist.stream().filter(t -> t.equals(NEOPLASM_ID)).findAny().isPresent()) {
```

Comment:
Could be else if instead of two separate statements :)

Reply:
generated code should be the same, but changing it :-0